module/zfs/vdev_disk.c

```diff
@@ -570,7 +570,7 @@ __vdev_disk_physio(struct block_device *bdev, zio_t *zio, caddr_t kbuf_ptr,
 	/* Extra reference to protect dio_request during vdev_submit_bio */
 	vdev_disk_dio_get(dr);
 	if (zio)
-		zio->io_delay = jiffies_64;
+		zio->io_disk_delay = gethrtime();
```

Comment:
We should pull this timing logic out of `vdev_disk.c` and up a level.  This has at least two major benefits.

1) It will then also work properly for file vdevs
2) It can be updated to include the `vdev_queue_aggregate()` time which in the current implementation isn't account for in the disk time.  That's vdev_queue_io_to_issue() to vdev_queue_io_done().